src/dbc.cpp

```diff
@@ -25,6 +25,120 @@ const auto unitPattern = "\"(.*)\""; // Random string
 const auto receiverPattern = "([\\w\\,]+|Vector__XXX)*";
 const auto whiteSpace = "\\s";
 
+enum VALToken { Identifier = 0, CANId, SignalName, Value, Description };
+
+struct VALObject {
+	uint32_t can_id;
+	std::string signal_name;
+	std::vector<libdbc::Signal::SignalValueDescriptions> vd;
+};
+
+bool parseVal(const std::string& str, VALObject& obj) {

```

Comment:
This should be a member on the class of signal not external imo. External classes shouldn't know of these details just that they get the members after. Also not a huge fan of how long this function is. Perhaps split this up into some sub functions? 

I am trying to follow what is going on and having a bit of a hard time. I will take another look again when I get a chance here.

Reply:
i don't think so, because this is related parsing the val object. It does the same thing like the regular expression, but in a different way because of the unknown numbers of value labels available. Therefore regex is not possible here